src/yacfg/yacfg.py

```diff
@@ -256,51 +251,52 @@ def generate_outputs(config_data, template_list, env, output_path=None):
     :type template_list: list[str]
     :param env: jinja2 template environment
     :type env: Environment
-    :param output_path: path where to generate_via_tuning_files output files,
+    :param output_path: path where to generate output files,
         or None to do a dry run
-    :type output_path: str
+    :type output_path: str | None
```

Comment:
There's some compatibility package to allow using | instead of Union[] in older Pythons. I think that it should be adopted in the future, so we can have modern type annotations.

Reply:
Yeah, I'm facing the same situation. I should search for an alternative tool that can analyze all the docstrings. Moreover, since we have typing, we can consider removing this information from the docstrings altogether.

Reply:
Nobody reads the generated docs anyways, I think it's ok not to have types there for now. It's IDE where it matters.

Sphynx needed this, iirc, https://pypi.org/project/sphinx-autodoc-typehints/. That jupyter book thing I brought in should work with it somehow, https://jupyterbook.org/en/stable/explain/sphinx.html

If you want to leave that up to me, I can try looking, since I'm who introduced jupyter book in first place